src/Server/Provision/ServerProvisioner.php

```diff
@@ -361,6 +386,59 @@ private function createTicket(array $params)
         ]);
     }
 
+    /**
+     * @param string $configID
+     * @param string $newKey
+     * @param bool $getNames    tells the function to return the frontend names that users see
```

Comment:
Boolean arguments are [usually bad](https://medium.com/@amlcurran/clean-code-the-curse-of-a-boolean-parameter-c237a830b7a3), as with this case. Better to have separate functions usually and have those functions share a common function call inside them for the parts that would be duplicated.